bin/data_frame_analyzer/Main.cc

```diff
@@ -101,6 +103,14 @@ int main(int argc, char** argv) {
     // TODO Actually use the specification to create the data frame and run the analysis
     ml::api::CDataFrameAnalysisSpecification dataFrameAnalysisSpecification{
         makeDataFrameAnalysisSpecification(configFile)};
+    if (dataFrameAnalysisSpecification.bad()) {
+        LOG_FATAL("Failed to parse analysis specification");
+        return EXIT_FAILURE;
+    }
+    if (dataFrameAnalysisSpecification.threads() > 1) {
+        ml::core::startDefaultAsyncExecutor(dataFrameAnalysisSpecification.threads() - 1);
```

Comment:
I'm not sure about the `- 1` here.  I get that it's for the main thread, but if we expect _either_ the main thread to be active _or_ the thread pool threads then it's not against the spirit of the `threads` argument for the thread pool to use as many threads as were specified.  Or am I missing something?

Reply:
The analyses are always launched in the background so the main thread stays live for control messages, monitoring, etc. This is managed by `CDataFrameAnalysisRunner`.

The rationale is if we've been told to use one thread (I don't count the mainly idle main thread) we might as well execute everything sequentially, since we don't then have any overhead of passing values or blocking for thread communication, i.e. see `CImmediateExecutor`.

Reply:
Makes sense.  Thanks.

Reply:
Actually, that makes sense for the `> 1` on line 110, but not the `- 1` on line 111.  If we're told to use 2 threads then as it stands we'll create a thread pool with 1 thread.